### ISSUE TITLE
Add configuration dump/restore commands for ARX configurations.

### ISSUE THREAD

**Issue by rh-codebase** [ovro-lwa/lwa-issues#155]
Saving and restoring the ARX configurations will help preserve settings and facilitate initializing new boards. Note that there is no fundamental way to dump the memory slots(0,1 or 2) containing stored configurations. Those must first be loaded into the running configuration and from there dumped to some YAML file.  

There are some edge cases to examine: 
1) Where to store a flag indicating the config was locally modified using the set_chan_cfg() type commands.  This flag will be set to 0 when either the load_cfg() command or restore_cfg() commands are used as they replace the entire config.
2) Should the existing save_cfg() command trigger a dump? Most likely yes as this will ensure a record exists of possible changes. OR
3) The save_cfg() could be removed from the external client API. Hence, memory slots can only be initialized through the new restore_cfg() API. That is, read from a file.
4) In addition, it may be handy to give these configurations names as there might be more than 3. The names would be used instead of memory slot numbers. If the config is already in one of the 3 slots, then it can be quickly loaded. Otherwise, it is read from a file with the option of storing in one of the memory locations for future reuse.

**Comment by ldaddario** [comment]
You're making this way more complicated than necessary.

What's needed is to think though a policy for how ARX configuratons should be managed.  Only when that's decided can you see whether some new code is needed to enforce/implement the policy.  Often nothing new is needed.  It's a huge mistake for someone to create code based on what he thinks the policy will be like.  It then becomes a _fait accompli_, making it hard to craft a policy that goes in a different direction.

The hardware feature that allows up to 3 configurations to be stored in non-volatile memory on the boards is just there as a convenience, in case it's useful.  It can be used or not.  Configurations can always be loaded via commands.  The on-board storage provides two capabilities beyond that, but these may or may not be needed:
1.  It's possible to switch configurations in a few ms, with a single command to all boards.  Loading each of the 704 channels separately by commands would take ~7s.  This is not useful if other parts of the system also need to be re-configured and they take longer than 7s.
2.  After a power cycle or reset, the configurations in slots 0 are automatically loaded.  If those slots contain the standard (commensal) configuration, then most likely no reconfiguration will be needed after a power cycle.

My opinion is that there should be one or more configurations (standard and special) that are always maintained by the observatory staff and cannot be changed by users.  They need to be adjusted for board changes, component aging, maybe seasonally (winter/summer), and if there is a policy change.  The master copies of these should be in files, not on the boards, where the files are read-only to most people.  If the on-boards slots are to be used, then there needs to be another policy about who can write what to each slot.  But if the slots are not operationally useful, then no policy is needed.  For example, you could decide that after a power cycle we won't rely on what's in slot 0 but instead will load everything from a file as part of a re-start script.

A totally separate issue (having little to do with use of the on-board slots) is raised by points 1 and 2 of this posting:  How can a user determine, for any time in the past during which he/she might have been observing, what configuration the ARX boards had?  If that feature is needed, then it's tricky to implement but it depends on what the policies say, so the policies still need to be decided first.  And it's equally tricky to implement for every other setting in the whole system, not just for the ARX boards.  The existence of the on-board slots does not make it harder, and it might (depending on policies) make it easier.

--Larry
